### PR TITLE
fix: update README.md with right docs link

### LINKED ISSUE #4: 404 documentation link
The documentation link in the readme markdown file leads to a 404 on the backstage.io website.

### PR DESCRIPTION
**Pull request by djamaile**
Should close: https://github.com/backstage/backstage-deploy/issues/4

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,4 +12,4 @@ npx backstage-deploy aws --stack backstage-demo
 
 ## Documentation
 
-- [Deploy documentation](https://backstage.io/docs/deployment/pulumi/aws)
+- [Deploy documentation]([https://backstage.io/docs/deployment/pulumi/aws](https://backstage.io/docs/deployment/backstage-deploy/aws-lightsail)https://backstage.io/docs/deployment/backstage-deploy/aws-lightsail)
```
Comment (mitchhentgesspotify) [inline README.md]: Wait, I think this borked - there's three links, and [the rendered markdown doesn't show a link at all anymore](https://github.com/backstage/backstage-deploy/pull/7/files?short_path=b335630#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5)

**Review by mitchhentgesspotify — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (djamaile) [reply README.md]: lmao, fat fingers. Updated it

File: README.md [hunk 2]
```diff
@@ -12,4 +12,4 @@ npx backstage-deploy aws --stack backstage-demo
 
 ## Documentation
 
-- [Deploy documentation](https://backstage.io/docs/deployment/pulumi/aws)
+- [Deploy documentation][https://backstage.io/docs/deployment/backstage-deploy/aws-lightsail]
```
Comment (mitchhentgesspotify) [inline README.md]: Unlucky :)

Screenshot 2023-07-10 at 10 48 27

```suggestion
- [Deploy documentation](https://backstage.io/docs/deployment/backstage-deploy/aws-lightsail)
```

**Review by mitchhentgesspotify — CHANGES_REQUESTED** [review]

**Review by mitchhentgesspotify — APPROVED** [review]
Let's roll 😎

**Pull request merged**